### DISCUSSION TITLE
Cpu vs gpu error?

### DISCUSSION THREAD

**Discussion by pylofe** [Crivella/ocr_translate#16] [Q&A] [answered]
Hey, so the application is working pretty good with the japanese manga type format so far, however I come to a problem.

It seems to be running cpu even though its the gpu version. I tried running the requirement cuda .txt and it says its installed already so I should have the right version unless I had some error. Is there a part in the code that decides this, I can try adding some print statements etc to see if its not using gpu properly?
![image](https://github.com/Crivella/ocr_translate/assets/149007083/fa3ec04c-462d-4453-a18d-3986dad4eeb6)

**Comment by Crivella** [comment, marked as answer]
Hi @pylofe ,

i would guess you need to set the environment variable `DEVICE="cuda"` [see the docs](https://crivella.github.io/ocr_translate/user/envs.html#app-variable-list) before launching the server

If that is not the case, you need to always install `requirements-torch-cuda.txt` first as pytorch uses the same package name under 2 different repository indexes for the CPU and GPU versions (once you installed one version, it will tell you it is installed regardless if you try to install the cuda on top of the cpu one)

Let me know if either of this solves it, in case i will move this to a discussion

> Is there a part in the code that decides this, I can try adding some print statements etc to see if its not using gpu properly?

That message is being written by easyocr, but at that point also other plugins that use pytorch are probably defaulting to CPU
If you want a more verbose output you can set the env variables `DJANGO_DEBUG="true"` and `DJANGO_LOG_LEVEL="debug"` before diving into manually adding print statements

**Comment by pylofe** [comment]
Great the environment variable DEVICE="cuda" did work, in task manager, gpu was finally being used.
Ill close this.
